src/main/java/frc/robot/classes/TimeOfFlightSensor.java

```diff
@@ -0,0 +1,227 @@
+      return data.length;
+    }
+
+    @Override
+    public char charAt(int index) {
+      return (char)data[index];
+    }
+
+    @Override
+    public CharSequence subSequence(int start, int end) {
+      return new String(data, start, end, StandardCharsets.UTF_8);
+    }
+
+  }
+
+  private static class IntRef {
+    int value;
+  }
+
+  int parseIntFromIndex(SingleCharSequence charSeq, int readLen, IntRef lastComma) {
+    int nextComma = 0;
+    try {
+      nextComma = findNextComma(charSeq.data, readLen, lastComma.value);
+      int value = Integer.parseInt(charSeq, lastComma.value + 1, nextComma, 10);
+      lastComma.value = nextComma;
+      return value;
+    } catch (Exception ex) {
+      return 0;
+    }
+  }
+
+  private int findNextComma(byte[] data, int readLen, int lastComma) {
+    while (true) {
+      if (readLen <= lastComma + 1 ) {
+        return readLen;
+      }
+      lastComma++;
+      if (data[lastComma] == ',') {
+        break;
+      }
+    }
+    return lastComma;
+  }
+
+  private final AtomicBoolean debugPrints = new AtomicBoolean();
+  private boolean hasDistance0;
+  private boolean hasDistance1;
+  private int distance0;
+  private int distance1;
+  private double lastReadTime;
+  private final ReentrantLock threadLock = new ReentrantLock();
+  private final Thread readThread;
+  private final AtomicBoolean threadRunning = new AtomicBoolean(true);
+
+
+  private void threadMain() {
+    // Using JNI for a non allocating read
+    int port = SerialPortJNI.serialInitializePort((byte)1);
+    SerialPortJNI.serialSetBaudRate(port, 115200);
+    SerialPortJNI.serialSetDataBits(port, (byte)8);
+    SerialPortJNI.serialSetParity(port, (byte)0);
+    SerialPortJNI.serialSetStopBits(port, (byte)10);
+
+    SerialPortJNI.serialSetTimeout(port, 1);
+    SerialPortJNI.serialEnableTermination(port, '\n');
+    
+    //int port = SerialPortJNI.serialInitializePort((byte)3);
+    HAL.report(tResourceType.kResourceType_SerialPort, 2);
+
+    byte[] buffer = new byte[257];
+    SingleCharSequence charSeq = new SingleCharSequence();
+    charSeq.data = buffer;
+    IntRef lastComma = new IntRef();
+
+    int distance0;
+    int distance1;
+
+    while (threadRunning.get()) {
+      int read = SerialPortJNI.serialRead(port, buffer, buffer.length - 1);
+      if (read <= 0) {
+        try {
+          threadLock.lock();
+          this.hasDistance0 = false;
+          this.hasDistance1 = false;
+        } finally {
+          threadLock.unlock();
+        }
+        continue;
+      }
+      if (!threadRunning.get()) {
+        break;
+      }
+
+      // Trim trailing newline if exists
+      if (buffer[read - 1] == '\n') {
+        read--;
+      }
+
+      if (read == 0) {
+        continue;
+      }
+
+      lastComma.value = -1;
+
+      distance0 = parseIntFromIndex(charSeq, read, lastComma);
+      distance1 = parseIntFromIndex(charSeq, read, lastComma);
+      boolean hasDistance0 = distance0 <= 0;
```

Comment:
Did I specify this wrong?  I must have.  My bad.  We want the distance to be invalid when 0 or -1.  For some reason we are reading 0 as a value.  And that should not be a valid sensor value.